src/OrasProject.Oras/Registry/Remote/Referrers.cs

```diff
@@ -105,4 +105,40 @@ internal static (IList<Descriptor>, bool) ApplyReferrerChanges(IList<Descriptor>
         
         return (updatedReferrers, updateRequired);
     }
+    
+    /// <summary>
+    /// IsReferrersFilterApplied checks if requstedFilter is in the applied filters list.
+    /// </summary>
+    /// <param name="appliedFilters"></param>
+    /// <param name="requestedFilter"></param>
+    /// <returns></returns>
+    internal static bool IsReferrersFilterApplied(string? appliedFilters, string requestedFilter) {
+        if (string.IsNullOrEmpty(appliedFilters) || string.IsNullOrEmpty(requestedFilter))
+        {
+            return false;
+        }
+
+        var filters = Strings.Split(appliedFilters, ",");
+        for (int i = 0; i < filters.Length; ++i)
+        {
+            if (filters[i] == requestedFilter)
+            {
+                return true;
+            }
+        }
```

Comment:
nit: Looks like `i` is not needed here, how about using `foreach`?

```suggestion
        foreach (var filter in filters)
        {
            if (filter == requestedFilter)
            {
                return true;
            }
        }
}
```